src/server/page_server.hpp

```diff
@@ -323,7 +323,15 @@ class page_server
     std::future<void> m_follower_disc_future;
     std::mutex m_follower_disc_mutex;
 
-    cubthread::entry_workpool *m_worker_pool; // a worker_pool to take some background jobs that needs a thread entry
+    /*
+     * A worker_pool to take some asynchronous jobs that need a thread entry.
+     * It's used by two server_request_responders for tran servers and follower servers,
+     * and some extra jobs like the catch-up.
+     *
+     * Note that server_request_responders have to be destroyed earlier since they ensure all jobs are done in their dtors.
```

Comment:
This comment makes half sense to me. Suggestion:
```suggestion
     * Note that both tran_server_responder and follower_responder hold references to the worker pool.
     * Thus, they must be destroyed prior to the worker pool - to avoid dangling references.
     * The responders ensure that all jobs are waited for completion in their dtors.
```